streams-tests/jvm/src/test/scala/zio/stream/ChunkSpec.scala

```diff
@@ -182,4 +183,10 @@ class ChunkSpec extends Specification with ScalaCheck {
     ((c1 == c2) && (c1.hashCode == c2.hashCode)) must beTrue
   }
 
+  def testzipAllWith = {
+    Chunk(1, 2, 3).zipAllWith(Chunk(3, 2, 1))(_ => 0, _ => 0)(_ + _) == Chunk(4, 4, 4)
+    Chunk(1, 2, 3).zipAllWith(Chunk(3, 2))(_ => 0, _ => 0)(_ + _) == Chunk(4, 4, 0)
+    Chunk(1, 2).zipAllWith(Chunk(3, 2, 1))(_ => 0, _ => 0)(_ + _) == Chunk(4, 4, 0)
```

Comment:
Note that only this line is actually being tested – only the value last of the last statement is returned and `==` is just a pure function here, not an assertion. You could try joining all the conditions with `&&` to test the first 2 equalities too.

Reply:
@Vilkina 

This means that instead of writing:

```scala
Chunk(1, 2, 3).zipAllWith(Chunk(3, 2, 1))(_ => 0, _ => 0)(_ + _) == Chunk(4, 4, 4)
    Chunk(1, 2, 3).zipAllWith(Chunk(3, 2))(_ => 0, _ => 0)(_ + _) == Chunk(4, 4, 0)
    Chunk(1, 2).zipAllWith(Chunk(3, 2, 1))(_ => 0, _ => 0)(_ + _) == Chunk(4, 4, 0)
```

You should instead write:

```scala
(Chunk(1, 2, 3).zipAllWith(Chunk(3, 2, 1))(_ => 0, _ => 0)(_ + _) == Chunk(4, 4, 4)) &&
(Chunk(1, 2, 3).zipAllWith(Chunk(3, 2))(_ => 0, _ => 0)(_ + _) == Chunk(4, 4, 0)) &&
(Chunk(1, 2).zipAllWith(Chunk(3, 2, 1))(_ => 0, _ => 0)(_ + _) == Chunk(4, 4, 0))
```

The reason is that in Specs2, the line `Chunk(...) == Chunk(..)` constructs a value. So you have 3 lines of code, each constructing a value:

```scala
<construct value 1>
<construct value 2>
<construct value 3>
```

But in Scala, when you define a method like so:

```scala
def method = {
  <construct value 1>
  <construct value 2>
  <construct value 3>
}
```

Then the first two values (1 and 2, in this example), will be "thrown away". The method will return only the final value (3).

We want to return a test value that combines all them together. We want _all_ tests to pass, so we use the "and" operator `&&`, which can take two test results and return another test result, which will succeed only if _both_ tests succeeded.